lib/core/include/irods/irods_load_plugin.hpp

```diff
@@ -42,20 +36,30 @@ namespace irods
 
         fs::path plugin_home;
 
-        rodsEnv env;
-        int status = getRodsEnv( &env );
-        if ( !status ) {
-            if ( strlen( env.irodsPluginHome ) > 0 ) {
-                plugin_home = env.irodsPluginHome;
+        // If we're the server, use the plugin directory property from server_config.json when defined.
+        // TODO Document this in docs.irods.org.
+        if (SERVER_PT == ::ProcessType || AGENT_PT == ::ProcessType) {
+            // TODO Document this.
+            // TODO Add new keyword variable for this.
+            if (irods::server_property_exists("plugin_directory")) {
+                plugin_home = irods::get_server_property<std::string>("plugin_directory");
+            }
```

Comment:
Handle TODOs.

Reply:
Which property name do we want to use?
- `plugin_directory`
- `plugin_home`

Before this commit, the server and the client relied on the irods_environment.json file and `plugin_home` to change the plugin directory.

If we make server_config.json use `plugin_directory`, the property names become different between the server and client.

Solutions:
- (a) Let the properties be different - doesn't matter as long as it's documented
- (b) Use `plugin_home` because that's already established
- (c) Deprecate `plugin_home` and allow `plugin_directory` to be used in the client as well

Reply:
I think I like c) best... 

`plugin_home` doesn't really convey anything on its own... is 'home' supposed to mean 'root'... as in, this is where all the plugins live/go?

Reply:
I like `plugin_directory` too, but I can also see `plugin_home` being used. The word, home, is used a lot to convey "the root directory for X" (e.g. CATALINA_HOME, JAVA_HOME, etc).

And the directory structure for irods plugins isn't flat.

With that said, I still think "directory" is clearer.

Reply:
agreed.

Reply:
Okay. I'll open an issue for the deprecation/removal of `plugin_home` and add support for `plugin_directory` to irods_environment.json.

Reply:
This PR addresses #8019.

Done.